ext/compressors/snappy/snappy_compress.c

```diff
@@ -108,91 +108,78 @@ wt_snappy_compress(WT_COMPRESSOR *compressor, WT_SESSION *session,
 	char *snapbuf;
 
 	/*
-	 * dst_len was computed in wt_snappy_pre_size, so we know it's big
-	 * enough.  Skip past the space we'll use to store the final count
-	 * of compressed bytes.
+	 * dst_len was computed in snappy_pre_size, so we know it's big enough.
+	 * Skip past the space we'll use to store the final count of compressed
+	 * bytes.
 	 */
 	snaplen = dst_len - sizeof(size_t);
 	snapbuf = (char *)dst + sizeof(size_t);
 
 	/* snaplen is an input and an output arg. */
 	snret = snappy_compress((char *)src, src_len, snapbuf, &snaplen);
 
-	if (snret == SNAPPY_OK) {
-		if (snaplen + sizeof(size_t) < src_len) {
-			*result_lenp = snaplen + sizeof(size_t);
-			*compression_failed = 0;
-
-			/*
-			 * On decompression, snappy requires an exact compressed
-			 * byte count (the current value of snaplen). WiredTiger
-			 * does not preserve that value, so save snaplen at the
-			 * beginning of the destination buffer.
-			 *
-			 * Store the value in little-endian format.
-			 */
+	if (snret == SNAPPY_OK && snaplen + sizeof(size_t) < src_len) {
+		*result_lenp = snaplen + sizeof(size_t);
+		*compression_failed = 0;
+
+		/*
+		 * On decompression, snappy requires an exact compressed byte
+		 * count (the current value of snaplen). WiredTiger does not
+		 * preserve that value, so save snaplen at the beginning of
+		 * the destination buffer.
+		 *
+		 * Store the value in little-endian format.
+		 */
 #ifdef WORDS_BIGENDIAN
-			snaplen = snappy_bswap64(snaplen);
+		snaplen = snappy_bswap64(snaplen);
 #endif
-			*(size_t *)dst = snaplen;
-		} else
-			/* The compressor failed to produce a smaller result. */
-			*compression_failed = 1;
+		*(size_t *)dst = snaplen;
 		return (0);
 	}
-	return (wt_snappy_error(compressor, session, "snappy_compress", snret));
+
+	*compression_failed = 1;
+	return (snret == SNAPPY_OK ?
+	    0 : snappy_error(compressor, session, "snappy_compress", snret));
 }
 
 /*
- * wt_snappy_decompress --
+ * snappy_decompression --
  *	WiredTiger snappy decompression.
  */
 static int
-wt_snappy_decompress(WT_COMPRESSOR *compressor, WT_SESSION *session,
+snappy_decompression(WT_COMPRESSOR *compressor, WT_SESSION *session,
     uint8_t *src, size_t src_len,
     uint8_t *dst, size_t dst_len,
     size_t *result_lenp)
 {
-	WT_EXTENSION_API *wt_api;
 	snappy_status snret;
-	size_t snaplen;
-
-	wt_api = ((SNAPPY_COMPRESSOR *)compressor)->wt_api;
 
 	/*
 	 * Retrieve the saved length, handling little- to big-endian conversion
 	 * as necessary.
 	 */
-	snaplen = *(size_t *)src;
+	src_len = *(size_t *)src;
```

Comment:
I'm not a fan of using the parameter for a different purpose.  It was telling us something real about how much data is safe for the function to use.  Now that information gets replaced with whatever happened to be in the on-disk buffer.